README.adoc

```diff
@@ -82,6 +84,47 @@ Usage:
 buildPlugin(platforms: ['linux'], jdkVersions: [7, 8], findbugs: [archive: true, unstableTotalAll: '0'], checkstyle: [run: true, archive: true])
 ----
 
+=== buildPluginWithGradle()
+
+Builds a Jenkins plugin using Gradle.
+The implementation follows the standard build/test/archive pattern. 
+The method targets compatibility with link:https://github.com/jenkinsci/gradle-jpi-plugin[Gradle JPI Plugin],
+and it may not work for other use-cases.
+
+==== Optional arguments
+
+* `repo` (default: `null`  inherit from Multibranch) - custom Git repository to check out
+* `failFast` (default: `true`) - instruct the build to fail fast when one of the configurations fail
+* `platforms` (default: `['linux', 'windows']`) - Labels matching platforms to
+  execute the steps against in parallel
+* `jdkVersions` (default: `[8]`) - JDK version numbers, must match a version
+    number jdk tool installed
+* `configurations`: An alternative way to specify `platforms`, `jdkVersions` (that can not be combined
+  with any of them)
+** Those options will run the build for all combinations of their values. While that is desirable in
+  many cases, `configurations` permit to provide a specific combinations of label and java/jenkins versions to use
+** It is also possible to use a  `buildPlugin.recommendedConfigurations()` method to get recommended configurations for testing. 
+Note that the recommended configuration may change over time, 
+and hence your CI may break if the new recommended configuration is not compatible
+
+[source,groovy]
+----
+buildPlugin(/*...*/, configurations: [
+  [ platform: "linux", jdk: "8"],
```

Comment:
Spaces are not consistent here. There is one space after `[` but no space before `]`. In the original example for `buildPlugin` it's consistent.

Reply:
```suggestion
  [ platform: "linux", jdk: "8" ],
```